### PR TITLE
NFT Storage is here

### PR DESCRIPTION
**Pull request by vikiival**
- :zap: added nft-storage worker
- :zap: implemented nft-storage
- :zap: use NFT-Storage for Basilisk
- :zap: should refresh token if user had legacy ESTUARY one
- :zap: RMRK is using Nft storage
- :zap: unique is using nft storage
- :zap: app is doing direct upload

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type

- [x] Bugfix
- [x] Feature
- [x] Refactoring

### What's new?

- [ ] PR closes #<issue_number>
- [ ] Requires deployment <rubick/magick_issue>
- [ ] <brief_description_of_what_I've_added>

### Before submitting Pull Request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

### CODE REVIEW

File: utils/nftStorage.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    const { status, data } = await api.post<StorageApiResponse>(
+      '/pinFile',
+      file,
+      {
+        headers: {
+          'Content-Type': file.type ? `${file.type}` : '*/*',
+        },
+      }
+    )
+    console.log('[NFT::STORAGE] Pin Image', status)
```
Comment (roiLeo) [inline utils/nftStorage.ts:67]:
```suggestion
    console.log('[NFT::STORAGE] Pin Image', status)
```

could we use consola 🥺 
+
phrasing
`[NFT::STORAGE] Pin File`

**Review by roiLeo — COMMENTED** [review]
`nftStorage` definitely looks like something that can be moved in services folder (ref #3310)

By the way I find something strange, `$nuxt.context` should be accessible from this file but it's not, do you have any idea why?

**Comment by yangwao** [comment]
right away seems it works well yet, immediately returns 500 but reloading right away wfm 

Screenshot 2022-07-07 at 10 53 40

**Comment by yangwao** [comment]
now it worked that returned metadata but no picture even after reload and w/o error, strange

image

**Comment by yangwao** [comment]
lmao, one can't simply think of this  

image

image

**Comment by yangwao** [comment]
Seems right away now it has worked, but doubt it could be the case every time. 

I would suggest it fetch it through pinata on the background one more time while minting has finished. 

And delay user going to fresh mint to like 5 seconds 

image

**Comment by petersopko** [comment]
minted collection and it's visible with name right away, seems to be closing 
- https://github.com/kodadot/nft-gallery/issues/3291
image

redirected to minting NFT after which minted NFT shows up right away:
image

https://deploy-preview-3358--koda-nuxt.netlify.app/bsx/gallery/2600576003-1

File: utils/nftStorage.ts (hunk 1 above)
Comment (yangwao) [reply utils/nftStorage.ts]: #3310 ?

**Comment by yangwao** [comment]
lgtm, works right away after mint. 
has someone tested it for rmrk?

**Comment by vikiival** [comment]
> has someone tested it for rmrk?

[Magick](https://kodadot.xyz/rmrk/gallery/13445851-8cc1b91e899d9be40e-2CENTS-MAGICK-0000000000000005)

**Comment by yangwao** [comment]
The only downside at the current implementation we are limited to 100MB per file until we rework it so that clients push it directly to NFT storage endpoint
- #3334 

image

**Pull request merged**